ls1intum/Artemis
# `General`: Filter list of submissions and complaints when accessing via the pie chart on exercise dashboard

### Checklist
#### General


#### Client

### Motivation and Context



### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- at least 2 Students
- 1x Modeling, Text and File Upload Exercise
- 1x Automatically assessed Programming exercise with complaints enabled
- 1x Programming exercise with manual assessment configured

1. Log in to Artemis
2. Make submissions for every prepared exercise and with both students
3. As instructor: Assess always one submission, leave the other submission unassessed
4. Navigate to the exercise assessment dashboard of every exercise
5. For the Modeling, Text, File Upload and manually assessed programming exercise: Validate that the pie chart shows the correct ratio of assessed and unassessed submissions.
6. Select one part in the pie itself. A new tab should open. Check that the displayed submissions match with the category you selected in the chart (e.g. only the assessed submission is shown if you selected this part in the chart).
7. Play around with the "show all" and "show locked" filters.
8. Press "reset filter" and ensure that all submissions are shown again, as well as the "show all" filter is selected
9. For the automatically assessed programming exercise: File a complaint about both submissions. Then, process only one of the complaints
10. Visit the exercise assessment dashboard and ensure that the correct ratio of assessed and unassessed complaints is shown.
11. Again, select a section in the chart. Ensure that the complaints are filtered accordingly in the new tab
12. If you select the part representing addressed complaints, make sure that the filter "Exclusively show addressed complaints" is selected in the complaint list
13. Again, play around with the filters and check that the reset mechanism works as expected. 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| assessment-filters.component.ts | 66.66% | 85.71% |
| list-of-complaints.component.ts | 77.01% | 83.92% |
| file-upload-assessment-dashboard.component.ts | 74.28% | 92.64% |
| modeling-assessment-dashboard.component.ts | 70.96% | 96.73% |
| programming-exercise-submissions.component.ts | 74.35% | 95% |
| abstract-assessment-dashboard.ts | 90.9% | 95.83% |
| exercise-assessment-dashboard.component.ts | 78.29% | 94.73% |
| text-assessment-dashboard.component.ts | 77.77% | 98.61% |
| navigation.utils.ts | 75.6% | 83.33% |


### Screenshots


Example appearance of the button that resets the filter applied through the chart:
![image](https://user-images.githubusercontent.com/80622272/175498072-9f318568-3c40-4d1b-b7f8-d756d28e2846.png)
